<launch>

    <arg name="load_gripper"                            default="true" />
    <arg name="robot_ip"                                default="172.16.0.2" />
    <arg name="table_height"                            default="0.12" />

    <arg name="start_realsense"                         default="true" />

    <arg name="rviz_config"                             default="rviz.rviz" />

    <arg name="enable_force_grasp"                      default="false" />

    <!-- Start robot controllers, moveit and rviz -->
    <include file="$(find panda_moveit_config)/launch/panda_control_moveit_rviz.launch">
        <arg name="load_gripper"                    value="$(arg load_gripper)" />
        <arg name="robot_ip"                        value="$(arg robot_ip)" />
        <arg name="rviz_command_args"               value="-d $(find panda_grasp_server)/rviz/$(arg rviz_config)" />
    </include>

    <!-- Start realsense, if necessary -->
    <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" if="$(arg start_realsense)">
        <arg name="align_depth"                         value="true"/>
        <arg name="enable_pointcloud"                   value="true"/>
    </include>

    <!-- Start the grasp and movement server node -->
    <node name="panda_grasp_server" pkg="panda_grasp_server" type="panda_grasp_server_node" output="screen">
        <param name="workspace/table_height"            value="$(arg table_height)" />
        <!-- Table size and bench size are supposed to be x,y,z -->
        <param name="workspace/table_size"              value="[1.0, 2.0, 0.8]" type="yaml" />
        <param name="workspace/bench_size"              value="[0.6, 0.6, 0.6]" type="yaml"/>
        <!-- Offset expressed in x,y coordinates with respect to the bench top surface center -->
        <param name="workspace/bench_mount_xy"          value="[0.1, 0.0]" type="yaml"/>
        <param name="planning/max_vel_scaling_factor"   value="0.3" />
        <param name="planning/max_acc_scaling_factor"   value="0.3" />
        <param name="planning/eef_link_id"              value="panda_tcp" if="$(arg load_gripper)"/>
        <param name="planning/eef_link_id"              value="panda_hand" unless="$(arg load_gripper)"/>
        <param name="ops_params/enable_force_grasp"     value="$(arg enable_force_grasp)"/>
    </node>

</launch>
